master/internal/stream/system.go

```diff
@@ -167,6 +187,33 @@ func (ps PublisherSet) Websocket(socket *websocket.Conn, c echo.Context) error {
 		streamer.Close()
 	}()
 
+	// detect permission changes
+	go func() {
+		reportProblem := func(ev pq.ListenerEventType, err error) {
+			if err != nil {
+				log.Errorf("reportProblem: %v\n", err.Error())
+			}
+		}
+		permListener := pq.NewListener(
+			// XXX: update this to use master config rather than hardcoded for a local db
+			"postgresql://postgres:postgres@localhost/determined?sslmode=disable",
+			minReconn,
+			maxReconn,
+			reportProblem,
+		)
+		// start listening for permission changes
+		err = permListener.Listen(permissionChannelName)
```

Comment:
This is going to connect one listener to the database per websocket.  If streaming updates and determined grow in popularity, that could be thousands of IPC connections.

Instead, can you model the justbootem detection as a member of the Publisher set or something, so there's only one connection to the database, and we can distribute the signal to all streaming clients in go code?